Sources/HTTP/HTTPHeadersAccessors.swift

```diff
@@ -0,0 +1,795 @@
+        let rangeStr = createRange(from: contentRange.lowerBound)
+        self.storage[.contentRange] = rangeStr.flatMap { [$0] }
+    }
+    #endif
+    
+    /// `Content-Type` header value
+    public var contentType: ContentType? {
+        get {
+            return self.storage[.contentType]?.first.flatMap { $0.components(separatedBy: ";").first.flatMap(ContentType.init(rawValue:)) }
+        }
+        set {
+            if let charset = self.storage[.contentType]?.first.flatMap({ parseParams($0)["charset"] }) {
+                self.storage[.contentType] = newValue.flatMap { ["\($0.rawValue); charset=\(charset)"] }
+            } else {
+                self.storage[.contentType] = newValue.flatMap { [$0.rawValue] }
+            }
+        }
+    }
+    
+    /// Extracted `charset` parameter in `Content-Type` header
+    public var contentCharset: String.Encoding? {
+        get {
+            return self.storage[.contentType]?.first.flatMap {
+                if let charset = parseParams($0)["charset"] {
+                    let cfEncoding = CFStringConvertIANACharSetNameToEncoding(charset as CFString)
+                    return String.Encoding(rawValue: CFStringConvertEncodingToNSStringEncoding(cfEncoding))
+                } else {
+                    return nil
+                }
+            }
+        }
+        set {
+            if let newValue = newValue {
+                let ianaEncoding = CFStringConvertEncodingToIANACharSetName(CFStringConvertNSStringEncodingToEncoding(newValue.rawValue)) as String
+                if self.storage[.contentType] != nil {
+                    self.storage[.contentType] = self.storage[.contentType]?.flatMap {
+                        let type = $0.components(separatedBy: ";").first?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "*"
+                        return "\(type); charset=\(ianaEncoding))"
+                    }
+                } else {
+                    self.storage[.contentType] = ["*; charset=\(ianaEncoding)"]
+                }
+                
+            } else {
+                self.storage[.contentType] = self.storage[.contentType]?.flatMap { ($0.components(separatedBy: ";").first?.trimmingCharacters(in: .whitespacesAndNewlines)) }
+            }
+        }
+    }
+    
+    /// `Date` header value
+    public var date: Date? {
+        get {
+            return self.storage[.date]?.first.flatMap(Date.init(rfcString:))
+        }
+        set {
+            self.storage[.date] = newValue.flatMap { [$0.format(with: .rfc1123)] }
+        }
+    }
+    
+    /// `ETag` header value
+    public var eTag: String? {
+        get {
+            return self.storage[.eTag]?.first
+        }
+        set {
+            if let newValue = newValue {
+                // eTag value must be a quoted string, we check only suffix to be compatible with weak eTags
+                let finalValue = newValue.hasSuffix("\"") ? newValue : "\"\(newValue)\""
```

Comment:
escaping?

Reply:
No, that adds a quotation to newValue if it doesn't have already. for weak etags